### ISSUE TITLE
print errors when using a saved model

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstanarm#122]
#### Summary:
@bgoodri I think adding the printing of the number of unconstrained parameters introduced bugin that it will lead to an error if printing a model that was saved and loaded. 
The issue is that in this case `is_sfinstance_valid` will be `FALSE` and therefore 
`object@.MISC$stan_fit_instance$num_pars_unconstrained()` will not be available.

#### Reproducible Steps:
```r
library(rstanarm)
fit <- stan_glm(mpg ~ wt, data = mtcars)
save(fit, file = "rstanarm-fit.rda")

rm(fit)
load("rstanarm-fit.rda")
print(fit)
```
Then it starts printing and eventually errors with
```r
Error in .local(object, ...) : 
  the model object is not created or not valid
Called from: .local(object, ...)
```

#### RStanARM Version:
2.12.2

#### R Version:
`3.3.1`

#### Operating System:
OS X 10.12

**Comment by Felipe-Osorio** [comment]
+1 from R version 3.2.3, RHEL 7.0, rstanarm 2.12.1

your explanation makes sense, so I won't repeat the error here. sadly fixing this is out of my ballpark https://github.com/alyst/rstan/blob/8474ac6772582c930e3d51d1f353195ddce2a6ed/rstan/rstan/R/stanfit-class.R#L710

**Comment by jgabry** [comment]
We'll fix it in the next release. Super easy to fix actually. The simplest
thing to do would probably be to copy this information into a slot or
attribute of the stanreg object. The problem now is that we're trying to
pull it from the stanfit object itself every time.

**Comment by jgabry** [comment]
Actually, I just fixed it now in be633454baeba6f846d55fa4023f3ef6bc775a5a
